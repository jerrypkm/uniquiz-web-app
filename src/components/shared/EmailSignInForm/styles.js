const styles = (theme) => ({
  main: {
    width: "auto",
    display: "block", // Fix IE 11 issue.
    marginLeft: theme.spacing(3),
    marginRight: theme.spacing(3),
    [theme.breakpoints.up(400 + theme.spacing(6))]: {
      width: 400,
      marginLeft: "auto",
      marginRight: "auto",
    },
  },
  card: theme.card,
  paper: {
    marginTop: theme.spacing(8),
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    padding: `${theme.spacing(2)}px ${theme.spacing(3)}px ${theme.spacing(
      3
    )}px`,
  },
  form: {
    width: "100%", // Fix IE 11 issue.
    marginTop: theme.spacing(),
    marginBottom: theme.spacing(4),
  },
  submit: {
    marginTop: theme.spacing(3),
  },
  error: {
    marginTop: theme.spacing(3),
  },
});

export default styles;
