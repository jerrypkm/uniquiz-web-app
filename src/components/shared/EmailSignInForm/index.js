import React from "react";
import Button from "@material-ui/core/Button";
import FormControl from "@material-ui/core/FormControl";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import Checkbox from "@material-ui/core/Checkbox";
import Input from "@material-ui/core/Input";
import InputLabel from "@material-ui/core/InputLabel";
import withStyles from "@material-ui/core/styles/withStyles";
import FormHelperText from "@material-ui/core/FormHelperText";
import SignUpLink from "components/shared/SignUpLink";
import PropTypes from "prop-types";
import { PasswordForgetLink } from "../PasswordForgot";
import styles from "./styles";
import MoTypography from "components/library/MoTypography";
import { SIGN_IN } from "constants/i18n";
import Paper from "@material-ui/core/Paper";

const EmailSignInForm = ({ classes, onSubmit, isInvalid, onChange, error }) => {
  return (
    <Paper className={classes.card}>
      <main className={classes.main}>
        <div className={classes.paper}>
          <MoTypography font="breeSerif" marginBottom="md" variant="h4">
            {SIGN_IN.CORE}
          </MoTypography>
          <form className={classes.form} onSubmit={onSubmit}>
            <FormControl margin="normal" required fullWidth>
              <InputLabel htmlFor="email">Email Address</InputLabel>
              <Input
                id="email"
                name="email"
                autoComplete="email"
                type="email"
                autoFocus
                onChange={onChange}
              />
            </FormControl>
            <FormControl margin="normal" required fullWidth>
              <InputLabel htmlFor="password">Password</InputLabel>
              <Input
                name="password"
                type="password"
                id="password"
                autoComplete="current-password"
                onChange={onChange}
              />
            </FormControl>
            <FormControlLabel
              control={<Checkbox value="remember" color="primary" />}
              label="Remember me"
            />
            <Button
              type="submit"
              fullWidth
              variant="contained"
              color="primary"
              className={classes.submit}
              disabled={isInvalid}
            >
              Sign in
            </Button>
            {error && (
              <FormHelperText
                className={classes.error}
                id="component-error-text"
                margin="dense"
                error
                focused
              >
                {error.message}
              </FormHelperText>
            )}
          </form>
          <PasswordForgetLink />
          <SignUpLink />
        </div>
      </main>
    </Paper>
  );
};

EmailSignInForm.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(EmailSignInForm);
