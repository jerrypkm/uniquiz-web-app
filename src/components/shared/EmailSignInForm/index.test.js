import React from "react";

import ShallowRenderer from "react-test-renderer/shallow";
import EmailSignInForm from "./index.js";

const renderer = new ShallowRenderer();

describe("EmailSignInForm Component", () => {
  test("should match the snapshot", () => {
    renderer.render(<EmailSignInForm />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
