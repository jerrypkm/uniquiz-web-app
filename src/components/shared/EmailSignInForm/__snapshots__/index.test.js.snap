// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailSignInForm Component should match the snapshot 1`] = `
<EmailSignInForm
  classes={
    Object {
      "card": "EmailSignInForm-card-2",
      "error": "EmailSignInForm-error-6",
      "form": "EmailSignInForm-form-4",
      "main": "EmailSignInForm-main-1",
      "paper": "EmailSignInForm-paper-3",
      "submit": "EmailSignInForm-submit-5",
    }
  }
/>
`;
